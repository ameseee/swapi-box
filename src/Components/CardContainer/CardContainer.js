import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import '../../index.css';

const CardContainer = ({ vehicles }) => {
  const vehicleValues = Object.values(vehicles);
  console.log(vehicleValues);

  return vehicleValues.map(vehicle => {
    console.log(vehicle);
    return <Card
      key={vehicle.url}
      name={vehicle.name}
      model={vehicle.model}
      passengers={vehicle.passengers}
      vehicleClass={vehicle.vehicleClass} />
  });
  //console.log('vehicles: ', Object.values(vehicleValues[0]));


  // return <Card vehicles={vehicles} />

  // const peopleKeys = Object.keys(people)

  // return peopleKeys.map((person) => {
  //   console.log('SPECIES INSIDE MAP: ', people[person].species);
  //   return ( <Card
  //     name={people[person].name}
  //     key={people[person].name}
  //     species={people[person].species}/>
  //   )
  //
  // })

};


// CardContainer.propTypes = {
//   cleanedSwapi: PropTypes.array
// };

export default CardContainer;
