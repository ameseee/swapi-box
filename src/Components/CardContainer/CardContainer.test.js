import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import config from '../../setupTests.js';

describe('<CardContainer />', () => {
  let wrapper = shallow(<CardContainer
    vehicles={{1: 'car', 2: 'truck'}}
    planets={{1: 'pluto', 2: 'earth'}}
    people={{1: 'Luke', 2: 'Leia'}}
    selected={'people'}
  />);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the Card snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
