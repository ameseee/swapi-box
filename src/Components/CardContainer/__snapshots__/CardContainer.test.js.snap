// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContainer /> should match the Card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    people={
        Object {
            "1": "Luke",
            "2": "Leia",
          }
    }
    planets={
        Object {
            "1": "pluto",
            "2": "earth",
          }
    }
    selected="people"
    vehicles={
        Object {
            "1": "car",
            "2": "truck",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Card
      0="L"
      1="u"
      2="k"
      3="e"
      toggleFavorited={undefined}
/>,
    <Card
      0="L"
      1="e"
      2="i"
      3="a"
      toggleFavorited={undefined}
/>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Card
        0="L"
        1="u"
        2="k"
        3="e"
        toggleFavorited={undefined}
/>,
      <Card
        0="L"
        1="e"
        2="i"
        3="a"
        toggleFavorited={undefined}
/>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
